### ISSUE TITLE
[Bug]: Get "Invalid Cluster Cursor Format" error when scanning if node name contains hyphen

### ISSUE THREAD

**Issue by tf-nphillips** [RedisInsight/RedisInsight#2865]
**Preconditions**
Redis cluster where node host names contain one or more hyphens (ie myredis-0002-001)
Note that AWS Elasticache Redis instances with cluster mode enabled use hyphens in the shard/node names.

**Steps to reproduce**
1. Connect to Redis cluster that contains node names with hyphens in them
2. Click "Scan more" after initial list of keys appears
3. Get "Invalid Cluster Cursor Format" error message

**Actual behavior**
The "Scan more" functionality posts a keys command to backend:

`POST http://localhost:5000/api/databases/{guid}/keys?encoding=buffer`
```
{
	"cursor": "myredis-cme-0002-001.myredis-cme.redishost.com:6379@26230||myredis-cme-0001-001.myredis-cme.redishost.com:6379@16950",
	"count": 10000,
	"type": null,
	"match": "*",
	"keysInfo": false
}
```

The backend returns a 500 error:

```
{
	"statusCode": 500,
	"message": "Incorrect cluster cursor format.",
	"error": "Internal Server Error"
}
```

The front end displays the "Incorrect cluster cursor format" error message and does not return more keys matching the specified pattern

**Expected behavior**
The "Scan more" functionality posts keys command to backend where node names containing hyphens pass the `CLUSTER_CURSOR_REGEX` test, `parseClusterCursor(...)` does not throw and api request properly returns more keys from the cursor which are displayed in front end. 

Changing the `CLUSTER_CURSOR_REGEX` to account for hyphens in node names allows for the expected behavior above:

`const CLUSTER_CURSOR_REGEX = /^(([a-z0-9.-])+:[0-9]+(@-?\d+)(?:\|{2}(?!$)|$))+$/;`

**Comment by ViktarStarastsenka** [comment]
hi @tf-nphillips , thanks for reporting the issue and adding the fix.
it will be available with 2.40 in late December

**Comment by ViktarStarastsenka** [comment]
2.40 has been released today.
Please re-open the ticket if the issue still persists.
